x/authority/types/authorizations_test.go

```diff
@@ -0,0 +1,306 @@
+				},
+			}},
+			expectedError: nil,
+		},
+		{
+			name: "validate failed with duplicate msg url with different policies",
+			authorizations: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupEmergency,
+				},
+			}},
+			expectedError: errors.Wrap(
+				types.ErrInvalidAuthorizationList,
+				fmt.Sprintf("duplicate message url: %s", "ABC")),
+		},
+		{
+			name: "validate failed with duplicate msg url with same policies",
+			authorizations: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			expectedError: errors.Wrap(
+				types.ErrInvalidAuthorizationList,
+				fmt.Sprintf("duplicate message url: %s", "ABC")),
+		}}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.authorizations.Validate()
+			require.ErrorIs(t, err, tc.expectedError)
+		})
+	}
+}
+
+func TestAuthorizationList_RemoveAuthorizations(t *testing.T) {
+	tt := []struct {
+		name                string
+		oldList             types.AuthorizationList
+		removeAuthorization types.Authorization
+		expectedList        types.AuthorizationList
+	}{
+		{
+			name: "remove authorization successfully",
+			oldList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+				{
+					MsgUrl:           "XYZ",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			removeAuthorization: types.Authorization{
+				MsgUrl: "ABC",
+			},
+			expectedList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "XYZ",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+		},
+		{
+			name: "do not remove anything if authorization not found",
+			oldList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			removeAuthorization: types.Authorization{
+				MsgUrl: "XYZ",
+			},
+			expectedList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.oldList.RemoveAuthorizations(tc.removeAuthorization)
+			require.Equal(t, tc.expectedList, tc.oldList)
+		})
+	}
+}
+
+func TestDefaultAuthorizationsList(t *testing.T) {
+	t.Run("default authorizations list", func(t *testing.T) {
+		var OperationalPolicyMessageList = []string{
```

Comment:
These are the variable I think it makes sense to define as global var

Reply:
Added the global variables for building the DefaultMsgList , but I would prefer to keep this in the test so that the test fails if any of the values gets changed